### DISCUSSION TITLE
guys pls help,

### DISCUSSION THREAD

**Discussion by LINKRINK01** [community/community#61812] [Codespaces] [answered]
### Select Topic Area

Question

### Body

container@pterodactyl~ Server marked as running...
Defaulting to user installation because normal site-packages is not writeable
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=15)")': /simple/requirements/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=15)")': /simple/requirements/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=15)")': /simple/requirements/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=15)")': /simple/requirements/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=15)")': /simple/requirements/
ERROR: Could not find a version that satisfies the requirement requirements (from versions: none)
ERROR: No matching distribution found for requirements
Traceback (most recent call last):
  File "bot.py", line 3, in <module>
    import discord
ModuleNotFoundError: No module named 'discord'
container@pterodactyl~ Server marked as offline...

**Comment by nirav-gajera** [comment, marked as answer]
The error you are getting is a **ModuleNotFoundError**, which means that Python cannot find the `discord` module. This could be because the module is not installed, or because it is installed in a location that Python cannot find.

The first thing you should do is check to make sure that the `discord` module is installed. You can do this by running the following command in your terminal:

```
pip install discord
```

If the module is not installed, this command will install it for you.

If the module is already installed, you can try to set the `PYTHONPATH` environment variable to the directory where the `discord` module is installed. You can do this by adding the following line to your `.bashrc` or `.zshrc` file:

```
export PYTHONPATH=$PYTHONPATH:~/path/to/discord
```

Once you have set the `PYTHONPATH` environment variable, you should be able to import the `discord` module without any errors.

If you are still getting the error after setting the `PYTHONPATH` environment variable, then there may be a problem with your Python installation. You can try reinstalling Python or updating to the latest version.

Here are the steps on how to fix the error:

1. Install the `discord` module using the `pip install discord` command.
2. Set the `PYTHONPATH` environment variable to the directory where the `discord` module is installed.
3. Reinstall Python or update to the latest version.

If you are still getting the error after following these steps, then you may need to contact Discord support for help.
